### PR TITLE
[ticket/16076] addFileFilter to check max file size per mime type

### PR DESCRIPTION
**Pull request by mrgoldy**
PHPBB3-16076

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16076

### CODE REVIEW

**Comment by EA117** [comment]
What will be the intended relationship between the per-Attachment Extension Group "Maximum file size" and the main ACP Attachment Settings "Maximum File Size"?

If the per-group value is set to zero, does this mean "I want to be limited by whatever the main ACP Attachment Settings maximum file size is set to"?  Or is setting a per-group value to zero an assertion that "even though other downloads may be limited, this particular Attachment Extension Group needs to be unlimited"?

The proposed change hasn't stopped passing _"max_file_size: '{FILESIZE}b'"_ in the plupload config, so I'm just trying to picture what the actual effect, interaction and intention will be.

**Comment by mrgoldy** [comment]
All files are limited to the general "maximum file size".
Per group you can lower the maximum file size.

The general one, is the 'overall' rule for attachment filesize, each and every attachment has to adhere to. Otherwise you will start working 'exceptions' and then you lose track rather quickly.

This also simply means, setting a group's maximum filesize to 0, that a group does not specific maximum filesize, other than the overall one.

The provides a linear way of thinking / managing: 
"overall" is the absolute limit and you can lower them per group.

File: phpBB/assets/javascript/plupload.js [hunk 1]
```diff
@@ -446,6 +446,44 @@ phpbb.plupload.fileError = function(file, error) {
...
 phpbb.plupload.initialize();
 
+/**
+ * Add a file filter to check for max file sizes per mime type.
+ */
+plupload.addFileFilter('mime_types_max_file_size', function(types, file, cb) {
+	if (file.size !== 'undefined') {
+		$(types).each(function(i, type) {
+			let extensions = [],
+				exts_array = type.extensions.split(',');
```
Comment (marc1706) [inline phpBB/assets/javascript/plupload.js]: We're using camelCase in JS
```suggestion
				extsArray = type.extensions.split(',');
```

File: phpBB/assets/javascript/plupload.js [hunk 2]
```diff
@@ -446,6 +446,44 @@ phpbb.plupload.fileError = function(file, error) {
 phpbb.plupload.uploader = new plupload.Uploader(phpbb.plupload.config);
 phpbb.plupload.initialize();
 
+/**
+ * Add a file filter to check for max file sizes per mime type.
+ */
+plupload.addFileFilter('mime_types_max_file_size', function(types, file, cb) {
```
Comment (marc1706) [inline phpBB/assets/javascript/plupload.js]:
```suggestion
plupload.addFileFilter('mime_types_max_file_size', function(types, file, callback) {
```
Please rename all occurences to `callback`. There is just no point in calling it `cb` unless one wants to reduce readability.

**Pull request merged**